virify.nf

```diff
@@ -477,11 +499,17 @@ workflow annotate {
           checkv_db
         )
 
-        viphos_annotations = annotation.out.map { _, __, annotations -> annotations }.collect()
-        taxonomy_annotations = assign.out.map { _, __, taxonomy -> taxonomy }.collect()
-        checkv_results = checkV.out.map { _, __, quality_summary, ___ -> quality_summary }.collect()
+        def i = 1 
+        def j = 1 
+        def k = 1 
+
+        // enumerate collected files to prevent file collision
+        viphos_annotations = annotation.out.map { _, __, annotations -> tuple(annotations, i++) }.collect(){it -> it[0]} //{ annotations, count -> "$annotations".replace('.', '_' + count + '.') }
+        taxonomy_annotations = assign.out.map { _, __, taxonomy -> tuple(taxonomy, j++) }.collect(){it -> it[0]} //{ taxonomy, count -> "$taxonomy".replace('.', '_' + count + '.') }
+        checkv_results = checkV.out.map { _, __, quality_summary, ___ -> tuple(quality_summary, k++) }.collect(){it -> it[0]} //{ quality_summary, count -> "$quality_summary".replace('.', '_' + count + '.') }
```

Comment:
This is to support the -list of assemblies, correct?

Reply:
When running on a list of samples this was my first fix to prevent the `write_gff` file collision, but I think [this](https://github.com/EBI-Metagenomics/emg-viral-pipeline/pull/128/files#diff-271fae29b461eeb8b8010c1b16ce3d1daf0eb82f7e7d98d63372a0fc9f6f3e6cR9) should be sufficient as well since then the input files are enumerated and have different names too. (I just forgot to remove this afterwards, will check again)

Reply:
I see. The change on the contigs.first() it a but more invovled as we need to .join() the assemblies with the corresponding annotations. Otherwise, the pipeline could mix results from different assemblies.

Reply:
Yes, true. Maybe its better then to keep the fix and handle the bug in a separate PR.